.github/actions/setupconda/action.yaml

```diff
@@ -4,40 +4,9 @@ description: 'Create conda environment for GitHub Action Job'
 runs:
   using: "composite"
   steps:
-    - name: Cache Conda Packages
-      id: cache-packages
-      uses: actions/cache@v3
-      env:
-        # Increase this value to reset cache if conda/conda-lock.yml has not changed
-        CACHE_NUMBER: 0
+    - uses: mamba-org/setup-micromamba@v1
       with:
-        path: ~/conda_pkgs_dir
-        key:
-          ${{ runner.os }}-conda-packages-${{ env.CACHE_NUMBER }}-${{ hashFiles('conda/conda-lock.yml') }}
-
-    - name: Setup Mambaforge
-      uses: conda-incubator/setup-miniconda@v2
-      with:
-        miniforge-version: 4.14.0-0
-        miniforge-variant: Mambaforge
-        use-mamba: true
-        auto-update-conda: false
-        activate-environment: hackweek
-        use-only-tar-bz2: true # IMPORTANT: This needs to be set for conda package cache to work properly!
-
-    - name: Cache Entire Conda Environment
-      id: cache-env
-      uses: actions/cache@v3
-      env:
-        CACHE_NUMBER: 0
-      with:
-        path: ${{ env.CONDA }}/envs
-        key:
-          ${{ runner.os }}-conda-environment-${{ env.CACHE_NUMBER }}-${{ hashFiles('conda/conda-lock.yml') }}
-
-    - name: Update Conda Environment
-      if: steps.cache-env.outputs.cache-hit != 'true'
-      env:
-        ENVFILE: conda/conda-${{ runner.os == 'Linux' && 'linux' || 'osx'  }}-64.lock.yml
-      shell: bash -l {0}
-      run: mamba env update -n hackweek -f $ENVFILE
```

Comment:
`mamba-org/setup-micromamba@v1` is much simpler and installs directly from `conda-lock.yml`, which means that the correct platform is automatically chosen